packages/protocol/contract_layout_layer1.md

```diff
@@ -74,15 +74,17 @@
 |-----------------------------+------------------------------------------------------+------+--------+-------+-------------------------------------------------------|
 | __gap                       | uint256[50]                                          | 251  | 0      | 1600  | ERC20Vault |
 |-----------------------------+------------------------------------------------------+------+--------+-------+-------------------------------------------------------|
-| bridgedToCanonical          | mapping(address => struct ERC20Vault.CanonicalERC20) | 301  | 0      | 32    | ERC20Vault |
+| solverConditionToSolver     | mapping(bytes32 => address)                          | 301  | 0      | 32    | ERC20Vault |
 |-----------------------------+------------------------------------------------------+------+--------+-------+-------------------------------------------------------|
-| canonicalToBridged          | mapping(uint256 => mapping(address => address))      | 302  | 0      | 32    | ERC20Vault |
+| bridgedToCanonical          | mapping(address => struct ERC20Vault.CanonicalERC20) | 302  | 0      | 32    | ERC20Vault |
```

Comment:
It seems the layout is not backward compatible.

Reply:
Ah, putting solver logic in another contract is not the way to go. I'll keep things within `ERC20Vault` itself.